fourm/pseudolabel_frames.py

```diff
@@ -2,46 +2,88 @@
 import os
 import cv2
 from ultralytics import YOLO
+import argparse
+import json
 
-SHARDS = "/cluster/work/cotterell/mm_swissai/datasets/hdvila/1000_hd_vila_shuffled/0000000000.tar"
-OUTPUT_DIR = "bbox-yolo/extracted_frames"
-LABELED_OUTPUT_DIR = "bbox-yolo/labeled_frames"
+# Set up argument parser
+parser = argparse.ArgumentParser(description='Process video frames with YOLO')
+parser.add_argument('--shards', type=str, required=True, help='Path to the tar file containing video shards')
+parser.add_argument('--nth_frame', type=int, default=30, help='Select every nth frame (default: 30)')
+parser.add_argument('--max_frames', type=int, default=None, help='Maximum number of frames to process (default: None, process all)')
+args = parser.parse_args()
+
+SHARDS = args.shards
+NTH_FRAME = args.nth_frame
+MAX_FRAMES = args.max_frames
+OUTPUT_DIR = "extracted_frames"
+LABELED_OUTPUT_DIR = "labeled_frames"
+JSON_OUTPUT_DIR = "json_output"
 
 # Ensure output directories exist
 os.makedirs(OUTPUT_DIR, exist_ok=True)
 os.makedirs(LABELED_OUTPUT_DIR, exist_ok=True)
+os.makedirs(JSON_OUTPUT_DIR, exist_ok=True)
 
 # Load the YOLO model
-model = YOLO('bbox-yolo/yolov8n.pt')  # pretrained YOLOv8n model
+model = YOLO('/cluster/work/cotterell/yemara/ml-4m/bbox-yolo/yolov8n.pt') # pretrained YOLOv8n model
 
 # Extract the tar file
 with tarfile.open(SHARDS, "r") as tar:
-    tar.extractall(path="bbox-yolo/extracted_files")
+    tar.extractall(path="extracted_files")
```

Comment:
+1 to Markus's comment -- better perhaps to do everything within a tempdir like this https://stackoverflow.com/questions/3223604/how-do-i-create-a-temporary-directory-in-python